tests/unit/test_io_hdf5_h5tools.py

```diff
@@ -150,6 +151,35 @@ def test_write_dataset_list_compress(self):
         self.assertEqual(dset.shuffle, True)
         self.assertEqual(dset.fletcher32, True)
 
+    @unittest.skipIf("lzf" not in h5py_filters.encode,
+                     "LZF compression not supported in this h5py library install")
+    def test_write_dataset_list_compress_lzf(self):
+        a = H5DataIO(np.arange(30).reshape(5, 2, 3),
+                     compression='lzf',
+                     shuffle=True,
+                     fletcher32=True)
+        self.io.write_dataset(self.f, DatasetBuilder('test_dataset', a, attributes={}))
+        dset = self.f['test_dataset']
+        self.assertTrue(np.all(dset[:] == a.data))
+        self.assertEqual(dset.compression, 'lzf')
+        self.assertEqual(dset.shuffle, True)
+        self.assertEqual(dset.fletcher32, True)
+
+    @unittest.skipIf("szip" not in h5py_filters.encode,
+                     "SZIP compression not supported in this h5py library install")
+    def test_write_dataset_list_compress_szip(self):
+        a = H5DataIO(np.arange(30).reshape(5, 2, 3),
+                     compression='szip',
+                     compression_opts=('ec', 16),
+                     shuffle=True,
+                     fletcher32=True)
+        self.io.write_dataset(self.f, DatasetBuilder('test_dataset', a, attributes={}))
+        dset = self.f['test_dataset']
+        self.assertTrue(np.all(dset[:] == a.data))
+        self.assertEqual(dset.compression, 'lzf')
```

Comment:
```suggestion
        self.assertEqual(dset.compression, 'szip')
```

Reply:
otherwise looks good!

Reply:
@bendichter good catch. This should be fixed now. I don't have SZIP installed so I unfortunately could not run that test case.